### ISSUE TITLE
ERROR: test_mem_overlap.test_diophantine_fuzz

### ISSUE THREAD

**Issue by charris** [numpy/numpy#6881]
Error on windows. @pv Probably need to disable test on windows  until randint gets fixed for bigger integers (#6824).
```
======================================================================
ERROR: test_mem_overlap.test_diophantine_fuzz
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Py\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\Py\lib\site-packages\numpy\core\tests\test_mem_overlap.py", line 123, in test_diophantine_fuzz
    b = rng.randint(-1, b_ub+2)
  File "mtrand.pyx", line 951, in mtrand.RandomState.randint (numpy\random\mtrand\mtrand.c:10704)
OverflowError: Python int too large to convert to C long

```

**Comment by gfyoung** [comment]
Just a comment: this is quite a strange error from what I understand, since that ```OverflowError``` issue is for any 32-bit Python distribution regardless of the OS (my added test in #6824 fails for the 32-bit machine on Travis without my changes, yet it does not fail ```test_diophantine_fuzz```).

**Comment by njsmith** [comment]
@gfyoung: This one is showing up on win64. The weird thing about win64 is that it's a 64-bit platform (so pointers and numpy array indices are 64-bits), but the C `long` is only 32-bits ("LLP64"). On every other platform we support, `long` is the same size as a pointer and the same size as an array index ("ILP32" or "LP64").

**Comment by gfyoung** [comment]
@njsmith : Ah, okay.  I figured there was something unusual about the platform compared to those on Travis.  Good to know!

**Comment by gfyoung** [comment]
@njsmith: It just occurred to me: do you think we can close this issue once that PR is merged (I include the issue number in my commits, and Appveyor will confirm the fix)?  Or should it say open until it been manually confirmed?

**Comment by charris** [comment]
Duplicate of #6812.
